docs/unicode.md

```diff
@@ -0,0 +1,68 @@
+# Macros And Unicode
+Macros are used for sending multiple keystrokes in a single action. This is useful for
+things like unicode input, sending strings of text, or other automation.
+
+## Basic Macros
+The most basic macro is send_string(). It can be used to send any typed keycode, including the return and tab key.
+```python
+from kmk.macros.simple import send_string
+
+WOW = send_string("Wow, KMK is awesome!")
+
+keymap = [...WOW,...]
+```
+# Unicode
+Before using unicode mode, you will need to set your platform. This can be done either of these ways.
+
+	unicode_mode = UnicodeModes.LINUX
+	OR
+	keymap = [...KC.UC_MODE_LINUX,...]
+
+
+### Unicode Modes:
+On Linux IBUS is required, and on Windows, requires [WinCompose](https://github.com/samhocevar/wincompose)
+- Linux : UnicodeModes.LINUX or UnicodeModes.IBUS
+- Mac:    UnicodeModes.MACOS or UnicodeModes.OSX or UnicodeModes.RALT
+- Windows: UnicodeModes.WINC
+
+A note for IBUS users on Linux. This mode is not enabled by default, and will need to be turned on for this to work.
+This works on X11, though if you are on Wayland, or in some GTK apps, it MAY work, but is not supported.
+
+	export IBUS_ENABLE_CTRL_SHIFT_U=1
+
+### Unicode Examples
+
+To send a unicode hex string, use unicode_codepoint_sequence()
+
+```python
+from kmk.macros.unicode import unicode_codepoint_sequence
+
+TABLE_FLIP = unicode_codepoint_sequence([
+	"28", "30ce", "ca0", "75ca","ca0", "29",
+	"30ce", "5f61", "253b", "2501", "253b",
+])
+
+keymap = [...TABLE_FLIP,...]
+```
+
+To send a simple unicode symbol
```

Comment:
Very critical note: emojis like this are actually only supported on Linux, because ibus doesn't restrict codepoint length to 4 (it restricts to like 6 or 8 I think). MacOS and (I believe) Windows only support 4-character hex codepoints, which explicitly precludes emojis right now (but works for things like table flips)

Reply:
I'd also love to show an example of an inline table flip here even though table flips were used above in the codepoint sequence.

Further, let's bump this section up above codepoints. Codepoints manually are such a fringe usecase I doubt they'll ever be used, but sending a unicode string is _awesome_ and something we should publicize.

Reply:
I would argue tho opposite. I have seen much more use with people asking for this to type other language characters such as german. In QMK, I've seen people tap dance the "u" key to "ü" on double tap for this reason. I am willing to change to the table flip however here, but that's regardless of order.

Reply:
Misunderstood. Fixing

Reply:
right, but this send string implementation works for German characters just as well as it does for table flips, without having to manually convert to `0252` :)

let's do the demo with umlaut U and/or table flip, and maybe keep an emoji example for Linux folks. And put this section above the manual-codepoint section.